### PR TITLE
MythTV 29.1 -> 30.0

### PR DESCRIPTION
**Pull request by LouisDK1**
Restoration of this pull request: https://github.com/NixOS/nixpkgs/pull/67191

#### Motivation for this change

https://github.com/NixOS/nixpkgs/issues/54596

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

Jonathan Rudenberg <jonathan@titanous.com>

### CODE REVIEW

**Comment by LouisDK1** [comment]
I'll have to path some unpure stuff starting at line 6669 to prevent the build from failing if build with xnvctrl enabled. Will look into this tomorrow.

**Comment by LouisDK1** [comment]
syntax error, unexpected '=', expecting ';', at /home/louis/nixpkgs/pkgs/applications/video/mythtv/default.nix at line 33.
Can someone help to discover what's wrong: https://github.com/LouisDK1/nixpkgs/blob/mythtv/pkgs/applications/video/mythtv/default.nix
I mean the ";" is at the end of the line and there shouldn't be more than one. Right?
I'm getting the same error while trying to build the package locally.

**Comment by LouisDK1** [comment]
I have to combine these two call packages in all-packages.nix:
mythtv = libsForQt5.callPackage ../applications/video/mythtv { };

mythtv = callPackage  ../applications/video/mythtv {    libXNVCtrl = linuxPackages.nvidia_x11.settings.libXNVCtrl;  };

However I don't know how to do this.

**Comment by jonringer** [comment]
` { libXNVCtrl = linuxPackages.nvidia_x11.settings.libXNVCtrl; };` is just an attribute set (think dictionary in other languages) that is being passed to a function. Just fill the parameter with the values you need.

```
mythtv = libsForQt5.callPackage ../applications/video/mythtv {
  libXNVCtrl = linuxPackages.nvidia_x11.settings.libXNVCtrl;
 };

**Comment by LouisDK1** [comment]
@jonringer I got new build errors. Kinda strange since I synced non-patched packages with the upstream branch.

error: anonymous function at /home/louis/nixpkgs/pkgs/development/libraries/qt-5/5.12/default.nix:17:1 called with unexpected argument 'fetchpatch', at /home/louis/nixpkgs/lib/customisation.nix:69:12

**Comment by LouisDK1** [comment]
I'll commit some final changes within an hour.

**Comment by LouisDK1** [comment]
I'll have to correct a patch.

**Comment by LouisDK1** [comment]
_building '/nix/store/z7p5knmsrif1l9zf1m46h91ng3zpjnnn-mythtv-30.0.drv'...
unpacking sources
unpacking source archive /nix/store/fkm3d0yvrd4cq5y0aam4lsm32j6hqf2z-source
source root is source/mythtv
patching sources
applying patch /nix/store/2hsa4z1kqsk53lcg405pj2hq47yxlkrn-004-exiv2.patch
patching file libs/libmythmetadata/imagemetadata.cpp
applying patch /nix/store/m6jzly65il7xkpdkh0h522y2p7fwj9gf-disable_xnvctrl_detection.patch
patching file configure
Hunk #1 FAILED at 6641.
1 out of 1 hunk FAILED -- saving rejects to file configure.rej
builder for '/nix/store/z7p5knmsrif1l9zf1m46h91ng3zpjnnn-mythtv-30.0.drv' failed with exit code 1_

I'm a bit unsure if I have disabled enough stuff in disable_xnvctrl_detection.patch - is enough to comment line from line 6644 (if enabled xnvctrl; then) or should I start at line 6640 (if enabled x11; then) ?

**Comment by LouisDK1** [comment]
@jonringer Hi. Could you look at this updated pull request? Thanks.

File: pkgs/applications/video/mythtv/default.nix [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
 , libpulseaudio, fftwSinglePrec , lame, zlib, libGLU_combined, alsaLib, freetype
-, perl, pkgconfig , libX11, libXv, libXrandr, libXvMC, libXinerama, libXxf86vm
-, libXmu , yasm, libuuid, taglib, libtool, autoconf, automake, file, exiv2
-, linuxHeaders, fetchpatch
+, perl, pkgconfig , libsamplerate, libbluray, lzo, libX11, libXv, libXrandr, libXvMC, libXinerama, libXxf86vm
+, libXmu , yasm, libuuid, taglib, libtool, autoconf, automake, file, exiv2, linuxHeaders
+, libXNVCtrl, enable-xnvctrl ? false
 }:
 
 stdenv.mkDerivation rec {
```
Comment (jonringer) [inline pkgs/applications/video/mythtv/default.nix]:
```suggestion
mkDerivation rec {
```

File: pkgs/applications/video/mythtv/default.nix [hunk 2]
```diff
@@ -1,40 +1,38 @@
 { stdenv, fetchFromGitHub, which, qtbase, qtwebkit, qtscript, xlibsWrapper
```
Comment (jonringer) [inline pkgs/applications/video/mythtv/default.nix]:
```suggestion
{ stdenv, mkDerivation, fetchFromGitHub, which, qtbase, qtwebkit, qtscript, xlibsWrapper
```

File: pkgs/applications/video/mythtv/patches/fix-os-detection.patch [hunk 3]
```diff
@@ -0,0 +1,12 @@
+--- a/configure	1970-01-01 01:00:01.000000000 +0100
++++ b/configure	2019-10-25 19:29:19.948540422 +0200
+@@ -6645,8 +6645,7 @@
+         case $target_os in
+             linux)
+                 # Bah. Suse linux doesn't have xnvctrl.
+-                . /etc/os-release
+-                case $ID in
++                case nix in
```
Comment (jonringer) [inline pkgs/applications/video/mythtv/patches/fix-os-detection.patch]: not sure how i feel about this one

**Review by jonringer — COMMENTED** [review]
some of these are not launching right, applying the changes should fix this, IIRC

```
nix-shell:/home/jon/.cache/nix-review/pr-71002]$ ./results/mythtv/bin/mythwelcome
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Aborted (core dumped)
```

File: pkgs/applications/video/mythtv/default.nix [hunk 4]
```diff
@@ -1,40 +1,38 @@
...
     freetype qtbase qtwebkit qtscript lame zlib xlibsWrapper libGLU_combined
-    perl alsaLib libpulseaudio fftwSinglePrec libX11 libXv libXrandr libXvMC
+    perl libsamplerate libbluray lzo alsaLib libpulseaudio fftwSinglePrec libX11 libXv libXrandr libXvMC
     libXmu libXinerama libXxf86vm libXmu libuuid taglib exiv2
-  ];
+  ] ++ stdenv.lib.optional enable-xnvctrl libXNVCtrl;
   nativeBuildInputs = [ pkgconfig which yasm libtool autoconf automake file ];
 
-  configureFlags = [ "--dvb-path=${linuxHeaders}/include" ];
+  configureFlags = [ "--dvb-path=${linuxHeaders}/include" ] ++ stdenv.lib.optionals (!enable-xnvctrl) [  "--disable-xnvctrl" ];
```
Comment (rycee) [inline pkgs/applications/video/mythtv/default.nix]:
```suggestion
  configureFlags =
    [ "--dvb-path=${linuxHeaders}/include" ]
    ++ stdenv.lib.optionals (!enable-xnvctrl) [  "--disable-xnvctrl" ];
```

File: pkgs/applications/video/mythtv/default.nix [hunk 5]
```diff
@@ -1,40 +1,38 @@
 { stdenv, fetchFromGitHub, which, qtbase, qtwebkit, qtscript, xlibsWrapper
 , libpulseaudio, fftwSinglePrec , lame, zlib, libGLU_combined, alsaLib, freetype
-, perl, pkgconfig , libX11, libXv, libXrandr, libXvMC, libXinerama, libXxf86vm
-, libXmu , yasm, libuuid, taglib, libtool, autoconf, automake, file, exiv2
-, linuxHeaders, fetchpatch
+, perl, pkgconfig , libsamplerate, libbluray, lzo, libX11, libXv, libXrandr, libXvMC, libXinerama, libXxf86vm
+, libXmu , yasm, libuuid, taglib, libtool, autoconf, automake, file, exiv2, linuxHeaders
+, libXNVCtrl, enable-xnvctrl ? false
```
Comment (rycee) [inline pkgs/applications/video/mythtv/default.nix]: I would suggest using camel case for variable that don't refer to packages. That is,
```suggestion
, libXNVCtrl, enableXnvctrl ? false
```

File: pkgs/applications/video/mythtv/default.nix (hunk 1 above)
Comment (LouisDK1) [reply pkgs/applications/video/mythtv/default.nix]: Thanks. Patched in my latest commit.

File: pkgs/applications/video/mythtv/patches/fix-os-detection.patch (hunk 3 above)
Comment (LouisDK1) [reply pkgs/applications/video/mythtv/patches/fix-os-detection.patch]: I've made another fix which might be a better way to patch it.

**Comment by LouisDK1** [comment]
@jonringer Hi. Could you look at this updated pull request yet again? Thanks.

File: pkgs/applications/video/mythtv/patches/fix-os-detection.patch (hunk 3 above)
Comment (rycee) [reply pkgs/applications/video/mythtv/patches/fix-os-detection.patch]: I think you can remove this file now?

**Comment by rycee** [comment]
@GrahamcOfBorg build mythtv

File: pkgs/applications/video/mythtv/disable-os-detection.patch [hunk 6]
```diff
@@ -0,0 +1,51 @@
+--- a/configure	1970-01-01 01:00:01.000000000 +0100
++++ b/configure	2019-10-26 11:54:01.920776490 +0200
+@@ -6642,29 +6642,29 @@
+     require libXinerama X11/extensions/Xinerama.h XineramaQueryExtension -lXinerama
+     require libXext "X11/Xdefs.h X11/Xlib.h X11/extensions/Xext.h" XMissingExtension -lXext
+     if enabled xnvctrl; then
+-        case $target_os in
+-            linux)
++#        case $target_os in
```
Comment (rycee) [inline pkgs/applications/video/mythtv/disable-os-detection.patch]: An option would be to simply delete the commented lines instead of commenting them out.

**Review by rycee — APPROVED** [review]
Builds OK for me and the binaries I tried worked fine for me. E.g. `mythwelcome`.

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executables seem to work

```
[2 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/71002
1 package were build:
mythtv
```
I like @rycee 's suggestion

**Comment by rycee** [comment]
Rebased to master in 2896f00181b6dd11c3075bfc5706f02beec1e9e3.

**Pull request closed without merging**